Add support for instant clones and documentation (#43)
chef/kitchen-vcenter: README.md

 - `poweron` - Power on the new virtual machine. Default: true
 - `vm_name` - Specify name of virtual machine. Default: `<suite>-<platform>-<random-hexid>`
 - `resource_pool` - Name of the resource pool to use when creating the machine. Will search first pool by default, can use value 'Resources' for none.

>  - `clone_type` - Type of clone, will default to "full" to create complete copies of template. Other values: "linked", "instant"

Reviewer: We usually specify the default value at the end of the line, e.g. "- Type of clone. Default: full (creates a copy of the template)." - @chef/docs-team will likely add a better suggestion here.

Author: That's a sensible suggestion, as I pulled out the detailed documentation of clone_type into a section by itself anyway.

>  - `clone_type` - Type of clone, will default to "full" to create complete copies of template. Other values: "linked", "instant"

Reviewer 2: ```suggestion
 - `clone_type` - Type of clone, use "full" to create complete copies of template. Values: "full", "linked", "instant". Default: "full"
```

## Clone types

### Clone type: full

This takes a VM or template, copies the whole disk and then boots up the machine. Default mode of operation.

### Clone mode: linked

> Instead of a full copy, this will use delta disks to speed up the cloning process and uses much less IO operations. The `template` parameter has to be a VM in this case, as a template is not usable. After creation of the delta disks, the machine is booted up and writes to its delta only.

Reviewer 2: ```suggestion
 Instead of a full copy, "linked" uses delta disks to speed up the cloning process and uses many fewer IO operations. After creation of the delta disks, the machine is booted up and writes only to its delta. Requires: `--template VM` 
```
I don't know if it is `--template VM` or `template: VM`

> Depending on the underlying storage system, performance might vary greatly compared to full clones.

Reviewer 2: ```suggestion
 Depending on the underlying storage system, performance may vary greatly compared to full clones.
```

### Clone mode: instant

> The instant clone feature has been available as VMFork in earlier vSphere versions, but without a proper public API. With version 6.7.0, this became an official feature. Instant clones work by not only using a delta disk like linked clones, but also share memory with the source machine. Due to this property, the new machines are already booted up after cloning.

Reviewer 2: ```suggestion
 The instant clone feature was as VMFork in earlier vSphere versions, but without a proper public API. With version 6.7.0, **this** became an official feature. Instant clones work by not only using a delta disk like linked clones, but also share memory with the source machine. Due to **this** property, the new machines are already booted up after cloning.
```
Does **this** refer to the API? Or to the way instant clones work? Each use of **this** points in a different direction.

Prerequisites:
- vCenter version 6.7.0 or higher
- vSphere Hypervisor (aka ESXi) version 6.7.0 or higher
- VMware Tools installed and running
- a running source virtual machine (`template` parameter)

Limitations:

> - new VM will always be on the same host due to memory sharing

Reviewer 2: ```suggestion
 - A new VM is always on the same host because of memory sharing
```

> - current driver only supports the "Frozen Source VM" workflow (as it is more efficient than "Running Source VM" version)

Reviewer 2: ```suggestion
 - The current driver supports only the "Frozen Source VM" workflow, which is more efficient than "Running Source VM" version
```

Freezing the source VM:
- Login to the machine
- Execute the freeze operation via `vmtoolsd -cmd "instantclone.freeze"`

> - The machine does not execute any CPU instructions from this point on

Reviewer 2: ```suggestion
 - The machine does not execute any CPU instructions after this point
```

> New clones will resume exactly at the frozen point in time and resume CPU activity automatically. In contrast to some early blog posts, they do not

Reviewer 2: ```suggestion
 New clones resume  from exactly the frozen point in time and also resume CPU activity automatically. In contrast to some early blog posts, they do not
```
